### PR TITLE
[Ide][Mac] Enable Alert sheeting on Mac

### PR DESCRIPTION
**Pull request by sevoku**
This patch enables alert dialog sheeting on Mac:

grafik

### CODE REVIEW

**Comment by VincentDondain** [comment]
Fixed by adding:

```
if (alert.Window.IsSheet && alert.Window.SheetParent != null)
    alert.Window.SheetParent.EndSheet (alert.Window);
```

right before `NSApplication.SharedApplication.AbortModal ();` in `MacAlertDialogHandler.cs`.

File: main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs [hunk 1]
```diff
@@ -149,14 +149,19 @@ public bool Run (AlertDialogData data)
...
+									NSApplication.SharedApplication.AbortModal ();
 							}
 						});
 					});
 				}
 				
 				if (!data.Message.CancellationToken.IsCancellationRequested) {
-					NSWindow parent = data.TransientFor ?? IdeApp.Workbench.RootWindow;
+					NSWindow parent = null;
+					if (IdeTheme.UserInterfaceTheme != Theme.Dark || MacSystemInformation.OsVersion < MacSystemInformation.HighSierra) // sheeting is broken on High Sierra with dark NSAppearance
```
Comment (Therzok) [inline main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs]: Did you mean `&&` instead of `||` here?

Comment (Therzok) [reply main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs]: No, ignore my comment.

Comment (sevoku) [reply main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs]: nono, it's true if macOS version is lower than `HighSierra` (-> always sheet), or the theme is not `Dark` (sheet only with light theme on High Sierra).

**Pull request merged**